### PR TITLE
Update to bevy 0.13

### PR DESCRIPTION
**Pull request by IceSentry**

### CODE REVIEW

**Review by bonsairobo — CHANGES_REQUESTED** [review]
There's actually a small amount of tech debt we can eliminate, as it's in scope for upgrading `bevy_mod_picking`. Can you delete the `ray_map` module and make sure the example still works?

See https://github.com/ForesightMiningSoftwareCorporation/bevy_fsl_box_frame/blob/7a2d5d551946ba17ffba4e24b6f7d5bb83386e73/src/lib.rs#L11

File: src/drag_face.rs [hunk 1]
```diff
@@ -105,9 +105,9 @@ pub(crate) fn drag_face(
 /// Find the closest pair of points `(p1, p2)` where `p1` is on ray `r1` and
 /// `p2` is on ray `r2`. Returns `(t1, t2)` such that `p_n =
 /// r_n.get_point(t_n)`.
-fn closest_points_on_two_rays(r1: &Ray, r2: &Ray) -> Option<(f32, f32)> {
+fn closest_points_on_two_rays(r1: &Ray3d, r2: &Ray3d) -> Option<(f32, f32)> {
```
Comment (bonsairobo) [inline src/drag_face.rs:105]: IDK why this change is happening, but can we maybe localize the `.xyz()` call to the caller of `closest_points_on_two_rays`?

File: src/drag_face.rs [hunk 2]
```diff
@@ -58,14 +58,14 @@ pub(crate) fn drag_face(
...
+            drag_ray: Ray3d {
                 origin: world_position,
-                direction: world_normal,
+                direction: Direction3d::new(world_normal).unwrap(),
             },
         });
     }
     for drag_end in drag_end_events.read() {
-        let Ok(mut frame) = box_frames.get_component_mut::<BoxFrame>(drag_end.target) else {
+        let Ok((mut frame, _)) = box_frames.get_mut(drag_end.target) else {
```
Comment (bonsairobo) [inline src/drag_face.rs:65]: Why did this change?

Comment (IceSentry) [reply src/drag_face.rs:65]: `get_component_mut()` was deprecated and is removed in 0.14. I'm not familiar as why that was done on the bevy side.

File: src/drag_face.rs (hunk 1 above)
Comment (IceSentry) [reply src/drag_face.rs:105]: This happens because bevy introduced a `Direction3d` type that wraps a `Vec3` for the ray direction. Calling `.xyz()` just gets the underlying `Vec3`. I just realized I can deref it to get the `Vec3` instead of using `.xyz()`, although, I would expect the compiler to optimize it the same way honestly.

**Comment by IceSentry** [comment]
I removed some unnecessary `direction.xyz()` and removed the `ray_map` module and everything seems to work.

**Review by bonsairobo — APPROVED** [review]
